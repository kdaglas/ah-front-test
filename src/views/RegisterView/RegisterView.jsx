import React, { Component } from 'react'
import RegsiterAction from '../../actions/registerUserActionCreator';
import {connect} from 'react-redux';
import Register from '../../components/Register/Register';
import socialSignUpAction from '../../actions/SocialSignupAction';

import './RegisterView.css'
import toastr from 'toastr'



export class RegisterView extends Component {
    state={
        username:'',
        email:'',
        password:'',
        usernameValid:false,
        loading:true
    }

    componentWillReceiveProps(nextProps){
        if(nextProps.registerState.status===200){
            toastr.info('successful login')
            nextProps.history.push('/')
        }else if(nextProps.registerState.status===400){
                toastr.warning(
                    nextProps.registerState.data.username)
                    toastr.warning(
                        nextProps.registerState.data.email)
                        toastr.warning(
        nextProps.registerState.data.password)
        }
    }
    

    HandleChange=(e)=>{
        this.setState({
            [e.target.name]:e.target.value
        })
       
    }
    Clicked=(e)=>{
        e.preventDefault()
        const {RegsiterAction} = this.props;
        RegsiterAction(this.state)
       
    }
    
    responseFacebook=(resp)=>{
        socialSignUpAction(resp.accessToken)

    }
    render(){
         
    return (
       
        <div>
        <Register Clicked={this.Clicked} 
        responseFacebook={this.responseFacebook}
        successOrError={this.state.usernameValid? "has-danger":"has-success"}
         HandleChange={this.HandleChange}/>

        </div>
        )
    }
}

export const mapStateToProps = (state) => {
    return {
        registerState: state.registerUserReducer,
        socailSignUpState:state.socialSignupReducer
    }
}

export default connect(mapStateToProps,{RegsiterAction,socialSignUpAction})(RegisterView)