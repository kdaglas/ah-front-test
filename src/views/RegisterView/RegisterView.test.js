import React from 'react';
import { shallow,mount } from 'enzyme';
import {RegisterView} from './RegisterView';

const props = {
    RegsiterAction: jest.fn(),
    
    
}

describe('test register', ()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<RegisterView {...props} />)
    });
    it('matches snapshot', ()=>{
        expect(wrapper).toMatchSnapshot()
       
    })

    it('calls click event for clicked',()=>{
        wrapper.instance().Clicked({preventDefault(){}})
        expect(props.RegsiterAction).toBeCalled()
    })

    // it('calls click event for Handle submit ',()=>{
    //     wrapper.instance().HandleChange(jest.fn(event=>{}))
    //     expect(props.setState).toBeCalled()
    // })

    
})