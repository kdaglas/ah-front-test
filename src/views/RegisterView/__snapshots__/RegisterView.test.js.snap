// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test register matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterView
    RegsiterAction={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Register
          Clicked={[Function]}
          HandleChange={[Function]}
          successOrError="has-danger"
        />,
        <t
          appId="367351517400934"
          callback={[Function]}
          cssClass="kep-login-facebook"
          fields="name,email,picture"
          size="metro"
          tag="button"
          textButton="Login with Facebook"
          typeButton="button"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "Clicked": [Function],
          "HandleChange": [Function],
          "successOrError": "has-danger",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appId": "367351517400934",
          "callback": [Function],
          "cssClass": "kep-login-facebook",
          "fields": "name,email,picture",
          "size": "metro",
          "tag": "button",
          "textButton": "Login with Facebook",
          "typeButton": "button",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Register
            Clicked={[Function]}
            HandleChange={[Function]}
            successOrError="has-danger"
          />,
          <t
            appId="367351517400934"
            callback={[Function]}
            cssClass="kep-login-facebook"
            fields="name,email,picture"
            size="metro"
            tag="button"
            textButton="Login with Facebook"
            typeButton="button"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "Clicked": [Function],
            "HandleChange": [Function],
            "successOrError": "has-danger",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appId": "367351517400934",
            "callback": [Function],
            "cssClass": "kep-login-facebook",
            "fields": "name,email,picture",
            "size": "metro",
            "tag": "button",
            "textButton": "Login with Facebook",
            "typeButton": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
